src/index.ts

```diff
@@ -82,6 +85,126 @@ export const AWS: SDK = new Proxy({} as any, {
+                    url.searchParams.set(
+                      `PublishBatchRequestEntries.${k}.${i}`,
+                      // `PublishBatchRequestEntries.${k}.${i + 1}`,
+                      // `PublishBatchRequestEntry.${i + 1}.${k}`,
+                      // `PublishBatchRequestEntries.${i + 1}.${k}`,
+                      v as any,
+                    );
+                  }
+                }
+              }
+            }
+            url.searchParams.set("Action", resolveAction(methodName));
+            url.searchParams.set("Version", "2010-03-31");
+
+            const response = await sendRequest(url, {
+              method: "POST",
+              body: JSON.stringify(input),
+              headers: {
+                // host is required by AWS Signature V4: https://docs.aws.amazon.com/general/latest/gr/sigv4-create-canonical-request.html
+                Host: url.host,
+                "Accept-Encoding": "identity",
+                "Content-Type": resolveContentType(className),
+                "X-Amz-Target": resolveXAmzTarget(className, methodName),
+                "User-Agent": "itty-aws",
+              },
+            });
+
+            const responseText = await response.text();
+            const parsedXml =
+              responseText !== "" ? parseXml(responseText) : undefined;
+            if (response.status < 200 || response.status >= 300) {
+              const errorXmlObject = (
+                (parsedXml?.children[0] as XmlElement).children.find(
+                  (child) =>
+                    child instanceof XmlElement && child.name === "Error",
+                ) as XmlElement
+              ).children as XmlElement[];
+              const errorCode = (
+                errorXmlObject.find((child) => child.name === "Code")
+                  ?.children[0] as XmlElement
+              ).text;
+              const errorMessage = (
+                errorXmlObject.find((child) => child.name === "Message")
+                  ?.children[0] as XmlElement
+              ).text;
+
+              if (errorCode) {
+                throw new AWSError(errorMessage, errorCode);
+              }
+            } else {
+              const xml = parseXml(responseText);
+              const response = xml.children.find(
+                (node) =>
+                  node instanceof XmlElement &&
+                  node.name.toLowerCase() ===
+                    `${methodName}Response`.toLowerCase(),
+              ) as XmlElement;
+              const result = response.children.find(
+                (node) =>
+                  node instanceof XmlElement &&
+                  node.name.toLowerCase() ===
+                    `${methodName}Result`.toLowerCase(),
+              ) as XmlElement | undefined;
+
+              if (result === undefined) {
+                return {};
+              }
+
+              const formatResultXmlElement = (element: XmlElement) => {
```

Comment:
is there a link to the schema we can include here for reference? 

How comprehensive is this passing logic?

Reply:
This is where I look for reference: https://docs.aws.amazon.com/sns/latest/api/API_Operations.html

Reply:
From what I've seen, there's two types of responses. The simple ones (eg, Publish):

```xml
<PublishResponse xmlns="https://sns.amazonaws.com/doc/2010-03-31/">
    <PublishResult>
        <MessageId>567910cd-659e-55d4-8ccb-5aaf14679dc0</MessageId>
    </PublishResult>
    <ResponseMetadata>
        <RequestId>d74b8436-ae13-5ab4-a9ff-ce54dfea72a0</RequestId>
    </ResponseMetadata>
</PublishResponse>
```

Which is most of the actions, and they always look like this:

```xml
<{action}Response>
  <{action}Result>
    <{key}>{value}</{key}>
  </{action}Result>
</{action}Response>
```

But there's also GetTopicAttributes which looks like this:

```xml
<GetTopicAttributesResponse xmlns="https://sns.amazonaws.com/doc/2010-03-31/">
    <GetTopicAttributesResult>
        <Attributes>
            <entry>
                <key>Owner</key>
                <value>123456789012</value>
            </entry>
            <entry>
                <key>Policy</key>
                <value>...</value>
            </entry>
            <entry>
                <key>TopicArn</key>
                <value>arn:aws:sns:us-east-2:123456789012:My-Topic</value>
            </entry>
        </Attributes>
    </GetTopicAttributesResult>
    <ResponseMetadata>
        <RequestId>057f074c-33a7-11df-9540-99d0768312d3</RequestId>
    </ResponseMetadata>
</GetTopicAttributesResponse> 
```

Maybe there's more types, though.